packages/kernel-5.15/latest-kernel-full-config.sh

```diff
@@ -0,0 +1,146 @@
+#
+
+# Cleanup all tmp files/directories
+cleanup() {
+    rm -rf "$tmpdir"
+}
+
+trap cleanup INT EXIT
+
+bail() {
+    if [[ $# -gt 0 ]]; then
+        >&2 echo "Error: $*"
+    fi
+    exit 1
+}
+
+# Function to display usage information
+usage() {
+    cat << EOF
+Usage: $(basename "$0") -r RPM_FILE
+
+Merge kernel configurations with the following parameters:
+    -r RPM_FILE    Path to RPM file
+    -h             Display this help message
+
+Dependencies:
+    - docker
+    - rpm2cpio
+    - cpio
+    - tar
+EOF
+}
+
+usage_error() {
+    >&2 usage
+    bail "$1"
+}
+
+check_dependencies() {
+    hash rpm2cpio || usage_error "DEPENDENCY ERROR: Please install rpm2cpio somewhere in your PATH"
+    hash cpio || usage_error "DEPENDENCY ERROR: Please install cpio somewhere in your PATH"
+    hash tar || usage_error "DEPENDENCY ERROR: Please install tar somewhere in your PATH"
+    hash docker || usage_error "DEPENDENCY ERROR: Please install docker somewhere in your PATH"
+}
+
+# expect $pwd to be packages/kernel-${kver}
+merge_kernel_configs() {
+    rpm_file=$1
+    kernel_path=$PWD
+
+    version="$(rpm --query --nosignature --queryformat '%{VERSION}' "${rpm_file}")"
+    readarray -t br_patches < <(find "${kernel_path}" -name "*.patch")
+
+    tmpdir=$(mktemp -d)
+    cp "$rpm_file" "$tmpdir/"
+    pushd "$tmpdir" || bail "Could not move around"
+
+    rpm2cpio "${rpm_file}" | cpio -iu {,./}linux-"${version}".tar {,.}config-x86_64 {,.}config-aarch64 {,./}"*.patch" {,./}kernel.spec
+    tar -xof linux-"${version}".tar; rm linux-"${version}".tar
+
+    # Find patch ordering based upstream SRPM and apply in that order
+    readarray -t patches < <(grep -P "^Patch\d+" kernel.spec | sort -n -k1.6 | grep -oP "^Patch\d+: \K.*\.patch$" kernel.spec)
+
+    pushd "linux-${version}" || bail "Could not move into linux-${version}"
+
+    # Patches from the upstream
+    for patch in "${patches[@]}"; do
+        patch -p1 <"../$patch"
+    done
+
+    # Patches from bottlerocket
+    for patch in "${br_patches[@]}"; do
+        patch -p1 <"$patch"
+    done
+
+    popd || bail "Could not move around - 'popd' failed. Lets stop before we break anything further."
+
+    for arch in "x86_64" "aarch64"; do
+
+        br_cfg="${kernel_path}/config-bottlerocket"
+        microcode_cfg="${kernel_path}/../microcode/config-microcode"
+        al_cfg="${PWD}/config-${arch}"
+        linux_src="${PWD}/linux-${version}"
+
+        pushd "linux-${version}" || bail "Could not move into linux-${version}"
+
+        if [ "${arch}" = "aarch64" ]; then
+            karch="arm64"
+            script_args=("../config-${arch}" "../config-bottlerocket")
+        elif [ "${arch}" = "x86_64" ]; then
+            karch="x86"
+            script_args=("../config-${arch}" "../config-microcode" "../config-bottlerocket")
+        fi
+
+        # mount config files and start the sdk docker container
+        docker run --rm \
+            -v "$br_cfg":/config-bottlerocket \
```

Comment:
nit: just for the sake of consistency, if `shellcheck` doesn't complain about it I'll, use `${}`.